src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs

```diff
@@ -2993,20 +2993,6 @@ private void OutputStartingBrace()
             }
         }
 
-        private CompilerResults FromFileBatch(CompilerParameters options, string[] fileNames)
-        {
-            if (options == null)
-            {
-                throw new ArgumentNullException(nameof(options));
-            }
-            if (fileNames == null)
-            {
-                throw new ArgumentNullException(nameof(fileNames));
-            }
```

Comment:
Why? The method is private and it's called only with already checked or newly created arguments?